controllers/chat/initChatFromPost.js

```diff
@@ -1,13 +1,142 @@
 const {responseSuccess} = require('../../utils/Responses');
 const {CHANNEL_TYPE} = require('../../helpers/constants');
 const UsersFunction = require('../../databases/functions/users');
 const Getstream = require('../../vendor/getstream');
 const BetterSocialConstantListUtils = require('../../services/bettersocial/constantList/utils');
-const PostAnonUserInfoFunction = require('../../databases/functions/postAnonUserInfo');
+const BetterSocialCore = require('../../services/bettersocial');
+const {generate_channel_id_for_anon_chat} = require('../../services/bettersocial/chat/allAnonChat');
+
+const sign_to_anon_post = async (
+  client,
+  userModel,
+  targetUserModel,
+  source_id,
+  anonUserInfoFromPost
+) => {
+  let channel = await ChatAnonUserInfo.findOne({
+    where: {
+      my_anon_user_id: targetUserModel.user_id,
+      target_user_id: userModel.user_id,
+      context: 'post',
+      initiator: userModel.user_id,
+      source_id: source_id
+    }
+  });
+  if (channel) {
+    return channel;
+  } else {
+    let channel_id = generate_channel_id_for_anon_chat(
+      targetUserModel.user_id,
+      userModel.user_id,
+      'post'
+    );
+    let new_channel = await ChatAnonUserInfo.create({
+      channel_id: channel_id,
+      my_anon_user_id: targetUserModel.user_id,
+      target_user_id: userModel.user_id,
+      anon_user_info_color_code: anonUserInfoFromPost.anon_user_info_color_code,
+      anon_user_info_color_name: anonUserInfoFromPost.anon_user_info_color_name,
+      anon_user_info_emoji_code: anonUserInfoFromPost.anon_user_info_emoji_code,
+      anon_user_info_emoji_name: anonUserInfoFromPost.anon_user_info_emoji_name,
+      context: 'post',
+      source_id: source_id,
+      initiator: userModel.user_id
+    });
+    return new_channel;
+  }
+};
+
+const anon_to_sign_post = async (client, userModel, targetUserModel, source_id) => {
+  let channel = await ChatAnonUserInfo.findOne({
+    where: {
+      my_anon_user_id: userModel.user_id,
+      target_user_id: targetUserModel.user_id,
+      context: 'post',
+      initiator: userModel.user_id
+    }
+  });
+  if (channel) {
+    return channel;
+  } else {
+    let channel_id = generate_channel_id_for_anon_chat(
+      userModel.user_id,
+      targetUserModel.user_id,
+      'post'
+    );
+    const emoji = BetterSocialConstantListUtils.getRandomEmoji();
+    const color = BetterSocialConstantListUtils.getRandomColor();
+    let new_channel = await ChatAnonUserInfo.create({
+      channel_id: channel_id,
+      my_anon_user_id: userModel.user_id,
+      target_user_id: targetUserModel.user_id,
+      anon_user_info_color_code: color.code,
+      anon_user_info_color_name: color.color,
+      anon_user_info_emoji_code: emoji.emoji,
+      anon_user_info_emoji_name: emoji.name,
+      context: 'post',
+      source_id: source_id,
+      initiator: userModel.user_id
+    });
+    return new_channel;
+  }
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Similar to the previous comment, the function `anon_to_sign_post` lacks error handling for the case when `ChatAnonUserInfo.create()` fails.
```diff
    let new_channel = await ChatAnonUserInfo.create({
      channel_id: channel_id,
      my_anon_user_id: userModel.user_id,
      target_user_id: targetUserModel.user_id,
      anon_user_info_color_code: color.code,
      anon_user_info_color_name: color.color,
      anon_user_info_emoji_code: emoji.emoji,
      anon_user_info_emoji_name: emoji.name,
      context: 'post',
      source_id: source_id,
      initiator: userModel.user_id
    });
+    } catch (error) {
+      console.error(`Failed to create channel: ${error}`);
+      throw error;
+    }
    return new_channel;
  }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->